operators/multiclusterobservability/pkg/rendering/renderer_mcoa_test.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+// Licensed under the Apache License 2.0
+
+package rendering
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	mcov1beta2 "github.com/stolostron/multicluster-observability-operator/operators/multiclusterobservability/api/v1beta2"
+	mcoconfig "github.com/stolostron/multicluster-observability-operator/operators/multiclusterobservability/pkg/config"
+	"github.com/stolostron/multicluster-observability-operator/operators/multiclusterobservability/pkg/rendering/templates"
+	templatesutil "github.com/stolostron/multicluster-observability-operator/operators/pkg/rendering/templates"
+	"github.com/stretchr/testify/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/ptr"
+	addonv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	kustomizeres "sigs.k8s.io/kustomize/api/resource"
+)
+
+func TestRenderMCOADeployment(t *testing.T) {
+	wd, err := os.Getwd()
+	assert.NoError(t, err)
+	templatesPath := filepath.Join(wd, "..", "..", "manifests")
+	t.Setenv(templatesutil.TemplatesPathEnvVar, templatesPath)
+
+	tmplRenderer := templatesutil.NewTemplateRenderer(templatesPath)
+	mcoaTemplates, err := templates.GetOrLoadMCOATemplates(tmplRenderer)
+	assert.NoError(t, err)
+
+	var dp *kustomizeres.Resource
+	for _, template := range mcoaTemplates {
+		if template.GetKind() == "Deployment" {
+			dp = template.DeepCopy()
+			break
+		}
+	}
+
+	mco := &mcov1beta2.MultiClusterObservability{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"cr-key": "cr-value",
+			},
+			Name: "multicluster-observability",
+		},
+		Spec: mcov1beta2.MultiClusterObservabilitySpec{
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			AdvancedConfig: &mcov1beta2.AdvancedConfig{
+				MultiClusterObservabilityAddon: &mcov1beta2.CommonSpec{
+					Replicas: ptr.To[int32](1),
+					Resources: &corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("100m"),
+							corev1.ResourceMemory: resource.MustParse("256M"),
+						},
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("200m"),
+							corev1.ResourceMemory: resource.MustParse("512M"),
+						},
```

Comment:
You can add the EphemeralStorage overwrite here; this would also fix the unit and sonar tests. Plus, does it make more sense to use non-default values here?
```suggestion
					Resources: &corev1.ResourceRequirements{
						Requests: corev1.ResourceList{
							corev1.ResourceCPU:    resource.MustParse("200m"),
							corev1.ResourceMemory: resource.MustParse("512Mi"),
						},
						Limits: corev1.ResourceList{
							corev1.ResourceCPU:              resource.MustParse("400m"),
							corev1.ResourceMemory:           resource.MustParse("1Gi"),
							corev1.ResourceEphemeralStorage: resource.MustParse("4Gi"),
						},

```